uisdk/src/test/java/tech/dojo/pay/uisdk/presentation/PaymentFlowViewModelTest.kt

```diff
@@ -47,6 +49,34 @@ internal class PaymentFlowViewModelTest {
     private val observePaymentIntent: ObservePaymentIntent = mock()
     private val fetchPaymentMethodsUseCase: FetchPaymentMethodsUseCase = mock()
     private val updatePaymentStateUseCase: UpdatePaymentStateUseCase = mock()
+    private val isSDKInitializedCorrectlyUseCase: IsSDKInitializedCorrectlyUseCase = mock()
+
+    @Test
+    fun `initialize view model with Success state from payment intent should call fetch payment methods `() =
+        runTest {
+            // arrange
+            val paymentIntentFakeFlow: MutableStateFlow<PaymentIntentResult?> =
+                MutableStateFlow(successPaymentIntent())
```

Comment:
Do you explicitly need a MutableStateFlow? I see you are using this for `observePaymentIntent.observePaymentIntent()` but we never should expose mutable flows